src/common/procedure/src/local.rs

```diff
@@ -126,6 +129,8 @@ pub(crate) struct ManagerContext {
     procedures: RwLock<HashMap<ProcedureId, ProcedureMetaRef>>,
     /// Messages loaded from the procedure store.
     messages: Mutex<HashMap<ProcedureId, ProcedureMessage>>,
+    /// Ids and finished time of finished procedures.
+    finished_procedures: RwLock<Vec<(ProcedureId, Instant)>>,
```

Comment:
Will `finished_procedures` keep growing, consuming too many memory?

Reply:
Yes, I have noticed this issue and replaced it with a VecDeque.